### DISCUSSION TITLE
Smartwatch BLE

### DISCUSSION THREAD

**Discussion by DeBelsch** [mKeRix/room-assistant#336] [Q&A]
Hello,

I really appreciate your work and I'm using it bluetooth classic and my smartphone and it works great. I bought myself now a smartwatch from fossil (the gen 5) with Bluetooth 4.2 LE. You wrote in the documentation that we can use smartwatches. I tried the ble integration and couldn't find the smartwatch. Did i understand something wrong?

Thanks

**Comment by mKeRix** [comment]
The Bluetooth Low Energy integration in room-assistant relies on the device broadcasting a static ID regularly via advertisement packages - not all devices do this. Some smart watches don't broadcast anything while they are connected to your phone, some only broadcast with random IDs ("privacy mode"), some have a toggle in the settings that allows you to switch this.

If you can't find the watch via Bluetooth Low Energy you can still try to use it with BT Classic though. I just did a quick search and it seems like your watch is running on Wear OS, on which you can apparently find the MAC by going to Settings > System > About and then tapping on Model twice. You can then put that into your BT Classic config like you did for your smartphone.

**Reply by wtreur** [reply]
I have the exact same watch and had the same issue with BLE. I tried your suggestion @mKeRix, but the watch only became visible (using the API) after I disconnected it from my phone first. However, after a restart of room-assistant it was gone again

**Reply by mKeRix** [reply]
Hm, this seems like an unfortunate limitation of the watch then, sorry to hear that. :/ I get why the manufacturers implemented it that way, since it isn't really made for tracking (and ideally nobody but you should be able to track it), but this is of course annoying.

Unfortunately Android Wear does not seem to support broadcasting LE beacons through an app either atm, not even with random IDs.

**Reply by zigt1301** [reply]
I'm having a similar problem with my Garmin Vivoactive 3 Music. It has BT-Classic for connecting with headphones but that connection is not on without trying to use the headphones.

However, BLE scanning does find it with an ID but when I try to whitelist the ID it is automatically changed to null. Presumably because the same ID was previously used in BT-Classic and is still saved somewhere in a configuration file.

**Reply by mKeRix** [reply]
What is changed to null - the whitelist or the tag ID within room-assistant? In general the configuration of these two devices does not touch each other and room-assistant also does not save an internal state on the filesystem either. It will only use the default options and the config files you point it to.

**Reply by zigt1301** [reply]
I'm not sure what the distinction is between the whitelist or tag ID, but I took some screenshots to try to show what is happening. The problem occurs both with and without a tag ID override. The MAC is 104e89658732

<img width="327" alt="Screenshot 2020-11-09 at 11 22 22" src="https://user-images.githubusercontent.com/15857773/98529540-0761b680-227e-11eb-8793-4d1ae0adada3.png">
In the above image you can see the that I have two entities in my BLE whitelist and the lower one of them is my Garmin Vivoactive 3. I also put in a tagOverride for clarity. This is before clicking Save. 

<img width="350" alt="Screenshot 2020-11-09 at 11 22 47" src="https://user-images.githubusercontent.com/15857773/98529767-53146000-227e-11eb-8935-0714bd6edc22.png">
After I click save and the Home Assistant Add-On restarts, the whitelist entity changes from the MAC to "null" and the tagOverride ID changes from the MAC to "infinity".

**Reply by mKeRix** [reply]
Oh interesting, that is a bug with how Home Assistant saves add-on configs then! room-assistant does not touch those parts at all, schema validation is turned off in the add-on and the file will only be read, never written to. The way you saved your config it should definitely work. Something that you can try to maybe get around this: could you wrap your IDs in single quotes? So:

```yaml
bluetoothLowEnergy:
  whitelist:
    - 'id'
  tagOverrides:
    'id':
      name: Garmin
```

**Reply by zigt1301** [reply]
Thank you for the suggestion. Putting the ID's in quotes definitely seems to help. The MAC is not removed. 
![Screenshot 2020-11-09 at 22 02 41](https://user-images.githubusercontent.com/15857773/98596157-52f47e80-22d7-11eb-90f6-1dc5c131a4f2.png)

Still can't track my Vivoactive 3 Music, though. But that's probably because it is not broadcasting BLE perpetually.
